distributed/executor.py

```diff
@@ -193,6 +193,22 @@ def release(self):
             self._cleared = True
             self.executor._dec_ref(tokey(self.key))
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state.update({'event': self.event.is_set(), 'status': self.status})
+        del state['executor']
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+        e = default_executor()
+        event = Event()
+        if state.pop('event'):
+            event.set()
+        e.futures[tokey(state['key'])] = {'event': event,
+                                          'status': state.pop('status')}
+        state['executor'] = e
```

Comment:
I see two things that might help avoid errors like this:

1.  Go through the actual process that users will go through in the test.  So rather than pass the name directly force yourself to query the scheduler for the names of the published datasets
2.  As suggested in https://github.com/dask/distributed/pull/453/files#r76600071 use and yield from the `.scheduler` rpc object rather than use the `._send_to_scheduler` method.  The `._send_to_scheduler` method is only really intended to be used by the computational functions that need high-volume fire-and-forget operation.  In this case we probably want to wait/block/yield until the dataset has been properly loaded up in the scheduler.